src/Audit.NET.AmazonQLDB/Providers/AmazonQldbDataProvider.cs

```diff
@@ -146,23 +147,25 @@ public override Task ReplaceEventAsync(object eventId, AuditEvent auditEvent)
         /// <param name="eventId">The event ID to retrieve. 
         /// Must be a Primitive, a AmazonQLDBEntry or an array of any of these two types. The first (or only) element must be the Hash key, and the second element is the range key.
         /// </param>
-        public override Task<T> GetEventAsync<T>(object eventId) => Task.FromResult(GetFromQldb<T>(eventId));
+        public override Task<T> GetEventAsync<T>(object eventId) => GetFromQldb<T>(eventId);
 
-        private T GetFromQldb<T>(object eventId) where T : AuditEvent
+        private async Task<T> GetFromQldb<T>(object eventId) where T : AuditEvent
         {
             var driver = QldbDriver.Value;
             var (insertDocumentId, tableName) = (ValueTuple<string, string>)eventId;
-            IResult selectResult = null;
-            driver.Execute(trx =>
+            IAsyncResult selectResult = null;
+            await driver.Execute(async trx =>
             {
-                selectResult = trx.Execute(
+                selectResult = await trx.Execute(
                     $@"SELECT e.*
                       FROM {tableName} AS e BY eid                      
                       WHERE eid = ?",
                     new ValueFactory().NewString(insertDocumentId));
             });
-
-            var selectedEvent = selectResult.First();
+            var enumerator = selectResult.GetAsyncEnumerator();
+            await enumerator.MoveNextAsync();
+            var selectedEvent = enumerator.Current;
+            await enumerator.DisposeAsync();
```

Comment:
We should put the enumerator within an await using block

Reply:
What this code does is to get the first element and close the stream. With await using we would have to iterate the entire stream.

Reply:
yes, but the difference is that the using statement will call the DisposeAsync even if an exception is thrown right?
Anyway we could simply call `FirstAsync()` method:

`var selectedEvent = await selectResult.FirstAsync();`